### PR TITLE
Separating Grid and Flex support for align-items

### PR DESCRIPTION
**Pull request by rachelandrew**
Further to emails with @chrisdavidmills and @wbamberg an attempt at adding a separate section for grid and flex layout support info.

I am sure this will need all kinds of changes but if we can get this one figured out then I can go through all the rest.

### CODE REVIEW

File: css/properties/align-items.json [hunk 1]
```diff
@@ -2,306 +2,372 @@
...
-          },
-          "status": {
-            "experimental": false,
-            "standard_track": true,
-            "deprecated": false
-          }
-        },
-        "first_last_baseline": {
+        "flex_context": {
+          "description": "Supported in Flex Layout",
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: This should be a sub-item of `__compat`.

File: css/properties/align-items.json [hunk 2]
```diff
@@ -2,306 +2,372 @@
...
-              "firefox_android": {
-                "version_added": "45"
-              },
-              "ie": {
-                "version_added": null
-              },
+              "chrome": [
+                {
+                  "version_added": "57"
+                }
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: Arrays are only permitted in support blocks when they have more than 1 entry.

File: css/properties/align-items.json [hunk 3]
```diff
@@ -2,306 +2,372 @@
...
+              ],
+              "chrome_android": [
+                {
+                  "version_added": "52"
+                }
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: ditto

File: css/properties/align-items.json [hunk 4]
```diff
@@ -2,306 +2,372 @@
...
+              ],
+              "edge": [
+                {
+                  "version_added": "16"
+                }
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: ditto

File: css/properties/align-items.json [hunk 5]
```diff
@@ -2,306 +2,372 @@
...
+              ],
+              "edge_mobile": [
+                {
+                  "version_added": true
+                }
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: ditto

File: css/properties/align-items.json [hunk 6]
```diff
@@ -2,306 +2,372 @@
...
+              ],
+              "firefox": [
+                {
+                  "version_added": "52"
+                }
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: ditto

File: css/properties/align-items.json [hunk 7]
```diff
@@ -2,306 +2,372 @@
...
+              ],
+              "firefox_android": [
+                {
+                  "version_added": "52"
+                }
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: ditto

File: css/properties/align-items.json (hunk 1 above)
Comment (rachelandrew) [reply css/properties/align-items.json]: I'm not sure I understand this comment - I was going on a suggestion by @wbamberg which had this outside __compat, so perhaps he could comment to clarify.

Comment (wbamberg) [reply css/properties/align-items.json]: It's my mistake: yes, `description` should be inside `__compat` :).

File: css/properties/align-items.json [hunk 8]
```diff
@@ -2,305 +2,357 @@
...
+                "version_added": "52"
               },
               "firefox_android": {
-                "version_added": "45"
-              },
-              "ie": {
-                "version_added": null
-              },
+                "version_added": "52"
+                },
```
Comment (ExE-Boss) [inline css/properties/align-items.json]: Wrong indentation

**Comment by wbamberg** [comment]
Thanks for this, @rachelandrew . It helps a lot to have something concrete to look at. With this data I think it would be best to include 2 tables in the page, with calls like:

```
<h2 id="Browser_compatibility">Browser compatibility</h2>

<h3 id="Support_in_Flex_layout">Support in Flex layout</h3>

<p>{{Compat("css.properties.align-items.flex_context")}}</p>

<h3 id="Support_in_Grid_layout">Support in Grid layout</h3>

<p>{{Compat("css.properties.align-items.grid_context")}}</p>
```

That would give us a table looking like this:

screen shot 2018-05-30 at 6 03 21 pm

I think this looks pretty clear. Note that this means the `description` fields are redundant, but I don't think they do any harm.

If we tried to include both in the same table I think it would get confusing when sub-sub-features like "<code>first baseline</code> and <code>last baseline</code>" are included:

screen shot 2018-05-30 at 8 11 07 pm

I can't think of any compelling reason not to use this as a model for the Grid/Flex/Multicol properties. I'm not sure if it will work very well when and if we want to support "aggregate" tables, like a table of all the Grid properties. But I don't think that should stop us moving forward with this.

I'd like to hear if @Elchi3 also thinks this is a good approach though, as it is a bit unusual.

File: css/properties/align-items.json [hunk 9]
```diff
@@ -2,305 +2,357 @@
...
-              },
-              "samsunginternet_android": {
-                "version_added": null
+          },
+          "first_last_baseline": {
+            "__compat": {
+              "description": "<code>first baseline</code> and <code>last baseline</code>",
+              "support": {
+                "webview_android": {
+                  "version_added": null
```
Comment (wbamberg) [inline css/properties/align-items.json]: It would be really nice to change some of these `null` values to Yes/No even if the version number is not known.

**Comment by rachelandrew** [comment]
I like the idea to have the separate tables. That looks nice and clear.

**Comment by ExE-Boss** [comment]
@rachelandrew @wbamberg We should probably also do the same table separation for [`gap`](https://developer.mozilla.org/docs/Web/CSS/gap), [`row-gap`](https://developer.mozilla.org/docs/Web/CSS/gap) and [`column-gap`](https://developer.mozilla.org/docs/Web/CSS/column-gap) (which is still duplicated in [`grid-column-gap`](https://developer.mozilla.org/docs/Web/CSS/grid-column-gap)), as a follow-up to #1990.

**Comment by rachelandrew** [comment]
@ExE-Boss yes that's the intention (I'm currently working on restructuring the Box Alignment documentation).

**Comment by Elchi3** [comment]
I also like the idea of separate tables. Great idea, @wbamberg :+1: 

> I'm not sure if it will work very well when and if we want to support "aggregate" tables, like a table of all the Grid properties.

Not sure about this yet either, but "aggregate" tables are a whole separate thing we'll have to look at on a future day. I agree to not worry about it for now.

**Comment by Elchi3** [comment]
So, this PR looks good to me, but unsure about whether there is more to come per the first comment
>  "if we can get this one figured out then I can go through all the rest."

If not, I'm happy to merge this and get it in today's release.

**Comment by rachelandrew** [comment]
This one stands alone, so lets publish this one and I'll work on the others doing an individual PR for each property if that makes sense? It will end up being a lot of files in one PR otherwise.

**Pull request merged**